relayer/src/keyring/store.rs

```diff
@@ -16,27 +17,45 @@ use k256::{
 };
 use serde_json::Value;
 use std::collections::BTreeMap;
+use std::convert::AsMut;
 use std::convert::TryFrom;
+use std::fs;
+use std::fs::File;
+use std::io::{Read, Write};
+use std::path::{Path, PathBuf};
 use std::str::FromStr;
+
 use tendermint::account::Id as AccountId;
 
+pub const KEYSTORE_DEFAULT_FOLDER: &str = ".rrly/keys/";
+pub const KEYSTORE_TEST_BACKEND: &str = "keyring-test";
+pub const KEYSTORE_FILE_EXTENSION: &str = "json";
+
 pub type Address = Vec<u8>;
 
 pub enum KeyRing {
-    MemoryKeyStore { store: BTreeMap<Address, KeyEntry> },
+    MemoryKeyStore {
+        store: BTreeMap<String, String>,
+        chain_config: ChainConfig,
```

Comment:
Do we really need to full chain config here? Can we instead just store the data that's actually needed from it (eg. the key name)?

Reply:
Actually I had it before with key name and chain id but it made calling the functions more complex and thought that this way was cleaner and also if we need more params in the config here in the future this would make it simpler to implement

Reply:
But I couldn't find a good pattern here. Ideally this is used as a property of the chain (`keybase`) so if the config could be "passed down" into the keybase then this `chain_config` property would not be needed.